README.md

```diff
@@ -150,6 +150,33 @@ all languages.
 
 ## Release Notes
 
+### Release 3.0.1 (April 24, 2024)
+* Upgraded amazon-kinesis-client from 2.4.4 to 2.5.8

```

Comment:
Is there a historic reason that we don't use PR links

Reply:
Not sure but also some of these are merged from multiple PRs if there are multiple version updates